projects/packages/my-jetpack/src/class-wpcom-products.php

```diff
@@ -35,20 +36,29 @@ class Wpcom_Products {
 	 * @return Object|WP_Error
 	 */
 	private static function get_products_from_wpcom() {
+		$connection        = new Connection_Manager();
+		$is_user_connected = $connection->is_user_connected();
+
+		// Request parameters.
+		$params = array(
+			'method'  => 'GET',
+			'headers' => array(
+				'X-Forwarded-For' => ( new Visitor() )->get_ip( true ),
+			),
+		);
 
-		$blog_id  = \Jetpack_Options::get_option( 'id' );
-		$endpoint = sprintf( '/sites/%d/products/?_locale=%s&type=jetpack', $blog_id, get_user_locale() );
-
-		$wpcom_request = Client::wpcom_json_api_request_as_blog(
-			$endpoint,
-			'1.1',
-			array(
-				'method'  => 'GET',
-				'headers' => array(
-					'X-Forwarded-For' => ( new Visitor() )->get_ip( true ),
-				),
+		// Hit the endpoint depending on the user's connection status.
```

Comment:
I think you mixed changes from the other PR here

Reply:
fixed. Thanks